### PR TITLE
Add prompt for variable creation

### PR DESCRIPTION
**Pull request by paulkaplan**
A simple first iteration of a new variable prompt. I wanted to get some feedback on the way this is implemented @rschamp. (The css definitely needs some work, and I avoided the header bar from the mockup because there isn't the data to fill it). 
_Update:_ Added the prompt header and refactored container/component divide.
screen shot 2017-04-26 at 10 17 26 am

Fixes https://github.com/LLK/scratch-blocks/issues/535

### CODE REVIEW

File: src/components/prompt/prompt.jsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    onCancel: React.PropTypes.func.isRequired,
+    onOk: React.PropTypes.func.isRequired,
+    placeholder: React.PropTypes.string
+};
+
+PromptComponent.defaultProps = {
+    placeholder: '...'
+};
+
+module.exports = PromptComponent;
```
Comment (rschamp) [inline src/components/prompt/prompt.jsx]: This should be split into a container and component.

Comment (paulkaplan) [inline src/components/prompt/prompt.jsx]: How do you mean? What part should be a container?

Comment (rschamp) [inline src/components/prompt/prompt.jsx]: The component should only call props for its handlers, and the container should implement them.

It doesn't matter too much at this point, other than for consistency, but as the components grow, it provides the split between the presentation and business logic.

**Comment by paulkaplan** [comment]
@rschamp thanks for the feedback. I separated the container/component boundary and updated the binding with scratch-blocks. I also added a header per the design spec. Should be ready for a real review!

**Pull request merged**